src/macaron/repo_finder/repo_finder_deps_dev.py

```diff
@@ -70,51 +72,54 @@ def find_repo(self, purl: PackageURL) -> str:
 
         return ""
 
-    def _create_urls(self, namespace: str, name: str, version: str, type_: str) -> list[str]:
+    def _create_urls(self, purl: PackageURL) -> list[str]:
         """
         Create the urls to search for the metadata relating to the passed artifact.
 
         If a version is not specified, remote API calls will be used to try and find one.
 
         Parameters
         ----------
-        namespace : str
-            The PURL namespace.
-        name: str
-            The PURL name.
-        version: str
-            The PURL version.
-        type : str
-            The PURL type.
+        purl : PackageURL
+            The PURL of an artifact.
 
         Returns
         -------
         list[str]
             The list of created URLs.
         """
-        base_url = self._create_type_specific_url(namespace, name, type_)
+        # See https://docs.deps.dev/api/v3alpha/
+        base_url = f"https://api.deps.dev/v3alpha/purl/{encode(purl.to_string()).replace('/', '%2F')}"
```

Comment:
You don't need to call `purl.to_string()` directly. You can either use the PURL in the format string or call `str()`. See https://github.com/oracle/macaron/pull/675